src/shims/unix/fs.rs

```diff
@@ -366,14 +367,33 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 return Ok(-1);
             }
         }
+
+        let o_nofollow = this.eval_libc_i32("O_NOFOLLOW");
+        if flag & o_nofollow == o_nofollow {
+            #[cfg(unix)]
+            {
+                use std::os::unix::fs::OpenOptionsExt;
+                options.custom_flags(o_nofollow);
+            }
+            // Strictly speaking, this emulation is not equivalent to the O_NOFOLLOW flag behavior,
+            // but it's good enough for Miri purposes.
+            #[cfg(not(unix))]
+            {
+                if path.is_symlink() {
```

Comment:
What will this do with a path like `/path/symlink-to-dir/file-in-dir`? AFAIK `is_symlink` only tests the top-level path, right? So this would have to check all prefixes of this path as well.

Reply:
Oh, reading the docs I learned O_NOFOLLOW also just checks the last component. Then yeah this emulation seems right.